web/src/main/java/org/springframework/security/web/authentication/ott/RedirectGeneratedOneTimeTokenSuccessHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2002-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.web.authentication.ott;
+
+import java.io.IOException;
+
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+import org.springframework.security.authentication.ott.OneTimeToken;
+import org.springframework.security.web.DefaultRedirectStrategy;
+import org.springframework.security.web.RedirectStrategy;
+import org.springframework.util.Assert;
+
+/**
+ * A {@link GeneratedOneTimeTokenSuccessHandler} that performs a redirect to a specific
+ * location
+ *
+ * @author Marcus da Coregio
+ * @since 6.4
+ */
+public final class RedirectGeneratedOneTimeTokenSuccessHandler implements GeneratedOneTimeTokenSuccessHandler {
```

Comment:
I'm not sure about a forward tho. If we forward from `POST /ott/generate` to `/login/ott` it will be a `POST` request, not a `GET`, I'm not sure how useful that is. Maybe the default implementation could store the `OneTimeToken` in the session and then redirect to the location.

Reply:
I think that the location that receives OTT would be in charge of doing something with it (e.g. sending it in an email, sending in a text message, placing it in session, etc) and then it would redirect to a new URL for the success page.

Reply:
Sure, we can continue the discussion on https://github.com/spring-projects/spring-security/issues/15623